app/models/disability_claim_document.rb

```diff
@@ -47,6 +47,16 @@ def ==(other)
     attributes == other.attributes
   end
 
+  # Redefining attributes method so that it picks up changes made after initialization
+  def attributes
+    Hash[attribute_set.map do |attribute|
+      [attribute.name.to_s, send(attribute.name)]
+    end]
+  end
```

Comment:
Are either of these preferable?

attribute_set.inject({}) do |attributes, attribute|
 attributes.merge( { attribute.name.to_s => send(attribute.name } )
end

Hash[attribute_set.map { |attr| [attr.name.to_s, send(attr.name)] }]